### PR TITLE
Configure module to be able to deploy all types of ECS services

### PR DESCRIPTION
**Pull request by bhavikkumar**
The changes on the module fixes issues #1  #2 and #3.

- Allows Fargate deployments
- Allows daemon services
- Allows ECS services without load balancers

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,57 +2,66 @@
...
-  service_name             = "${var.service_name}
-  splunk_token             = "${var.splunk_token}
-  splunk_url               = "${var.splunk_url}
-  spring_profile           = "default,${terraform.workspace}"
-  task_role_arn            = "${aws_iam_role.task.arn}"
+  service_name             = var.service_name
+  task_definition_arn      = aws_ecs_task_definition.bar.arn
   vpc_id                   = "vpc-124552462"
-  tags                     = "${merge(local.common_tags, var.tags)}"
+  tags                     = merge(local.common_tags, var.tags)}
```
Comment (alphafoobar) [inline README.md]: I notice that we have a mixture of syntax for `tags`, typically we see 

```
"${merge(local.common_tags, var.tags)}"
```

Is the syntax here correct? There also appears to be trailing `}` on the edited line.

File: README.md [hunk 2]
```diff
@@ -2,57 +2,66 @@
...
+  cluster_name             = "ecs-cluster-name"
+  external_lb_listener_arn = "arn:aws:elasticloadbalancing:region:account-id:listener/app/extlb"
+  external_lb_name         = "extlb"
+  internal_lb_listener_arn = ""
+  internal_lb_name         = ""
+  is_exposed_externally    = true
+  priority                 = 5
+  service_name             = var.service_name
+  task_definition_arn      = aws_ecs_task_definition.bar.arn
+  vpc_id                   = "vpc-124552462"
```
Comment (alphafoobar) [inline README.md:60]: Not newly introduced by this PR, but should we parameterize the `vpc_id`?

**Review by alphafoobar — COMMENTED** [review]
Made two observations for review. Please confirm before proceeding.

File: README.md (hunk 1 above)
Comment (bhavikkumar) [reply README.md]: Both are valid for us as we use HCL2. However we should be using the HCL2 way for consistency. Our  modules are now only compatible with Terraform 0.12.6 or higher. 

The trailing } is a error. I will push a fix.

File: README.md (hunk 2 above)
Comment (bhavikkumar) [reply README.md:60]: It is. 

The example usage just shows the minimal inputs required. It is up to the user of the module to provide these. How they provide them is not a concern for the module.

Comment (alphafoobar) [reply README.md:60]: Awesome - my bad! Reviewing before espresso I didn't notice the file was a `README`!

**Review by alphafoobar — APPROVED** [review]

**Pull request merged**